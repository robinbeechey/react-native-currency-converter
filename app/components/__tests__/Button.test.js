import React from 'react';
import renderer from 'react-test-renderer';
import { ClearButton, styles } from '../Button';
import buildStyles from '../../config/styles';

beforeAll(()=> {
    buildStyles();
});

it('renders succesfully', () => {
    const rendered = renderer.create(<ClearButton />).toJSON();
    expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
    expect(typeof styles).toBe('object');
});

it('renders custom text passed via props', () => {
    const rendered = renderer.create(<ClearButton text="hey this is a test"/>).toJSON();
    expect(rendered).toMatchSnapshot();
});