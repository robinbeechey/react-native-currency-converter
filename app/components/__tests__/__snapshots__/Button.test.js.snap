// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom text passed via props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "marginRight": 9,
          "width": 19,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "300",
          "paddingVertical": 20,
        }
      }
    >
      hey this is a test
    </Text>
  </View>
</View>
`;
